client/state/happychat/reducer.js

```diff
@@ -0,0 +1,82 @@
+import { combineReducers } from 'redux';
+import get from 'lodash/get';
+import find from 'lodash/find';
+import concat from 'lodash/concat';
+
+import {
+	HAPPYCHAT_SET_MESSAGE,
+	HAPPYCHAT_RECEIVE_EVENT,
+	HAPPYCHAT_SET_AUTOSCROLL,
+	HAPPYCHAT_CONNECTING,
+	HAPPYCHAT_CONNECTED,
+	HAPPYCHAT_CLOSING,
+} from 'state/action-types';
+
+const debug = require( 'debug' )( 'calypso:happychat:reducer' );
+
+const available = ( state = true ) => state;
+
+const timeline_event = ( state = [], action ) => {
+	switch ( action.type ) {
+		case HAPPYCHAT_RECEIVE_EVENT:
+			const event = action.event;
+			return [
+				event.message,
+				Object.assign( {}, {
+					id: event.id,
+					nick: event.user.nick,
+					image: event.user.picture,
+					user_id: event.user.id,
+					type: event.type,
+					links: get( event, 'meta.links' )
```

Comment:
It's not totally clear why we're storing this as an array instead of just an object.